### ISSUE TITLE
Draftail editor allows to link to root page rendering page non-functional

### ISSUE THREAD

**Issue by bentrm** [wagtail/wagtail#4605]
### Issue Summary

The following has come to my attention as a user has been complaining about a non-functional editing interface for a seemingly random sub-page.  The Draftail rich text editor allows to link to the root page of the installation:

![bildschirmfoto 2018-06-13 um 15 06 05](https://user-images.githubusercontent.com/1186959/41353073-6ea89ae8-6f1b-11e8-8b26-5249b40f3ff1.png)

Consequently, a non-functional link is insert as follows:

![bildschirmfoto 2018-06-13 um 14 45 46](https://user-images.githubusercontent.com/1186959/41352571-1f4069fa-6f1a-11e8-992e-f73a66a0f337.png)

Saving this page renders an error - nevertheless the page seems to be persisted. Its impossible to edit the page thereafter.

### Steps to Reproduce

1. Start a new project with `wagtail start myproject`
2. Edit models.py to include a page model with a RichTextField with the `link` feature enabled.
3. Add a link to the fields text clicking the "Root" page in the "Link"-popup.

I'm not entirely sure I'm using the feature correctly. Is it possible to reduce the available pages in the richtexts link popup to not include the root page?

### Technical details

* Python version: 3.6.5
* Django version: Look in your requirements.txt, or run `2.0.2`.
* Wagtail version: 2.1
* Browser version: Safari 11.1.1

**Comment by thibaudcolas** [comment]
Hey @bentrm, thank you for taking the time to review this.

I think this is because of the `can_choose_root` parameter defined where Draftail loads the link chooser:

https://github.com/wagtail/wagtail/blob/cab90e5d1b7ba61ff8da1ca372e595187bec301a/client/src/components/Draftail/sources/ModalWorkflowSource.js#L31-L37

I can see this call is done differently with the previous editor: 

https://github.com/wagtail/wagtail/blob/c7a418b1c360204ab8393386a2dd72312e21993a/wagtail/admin/static_src/wagtailadmin/js/page-chooser.js#L14-L16

I imagine the new definition with `can_choose_root=false` might clash with this server-side code: 

https://github.com/wagtail/wagtail/blob/d483c8d46543cd75caac8d8b4067173646072c81/wagtail/admin/views/chooser.py#L52-L60

…where a string of value `false` will be truthy. 

---

Looking at the history of this code, setting `can_choose_root: 'false',` dates back to the early developments of the new editor. I can't tell exactly what are the implications of this, whether we need to remove this parameter or instead support it like the previous editor does.

**Comment by gasman** [comment]
`can_choose_root` should just be omitted from the `urlParams` dict, I think - `can_choose_root=False` is the default behaviour of the page chooser (as per [this line](https://github.com/wagtail/wagtail/blob/d483c8d46543cd75caac8d8b4067173646072c81/wagtail/admin/views/chooser.py#L107)).

While the ability to choose the root is indeed an error, I'm more surprised that doing so causes the page to break completely, rather than just being treated as a broken link like any other page without a routable URL... on closer inspection, that's because we call `page.get_parent().id` without a check for the parent being `None`. We should fix that too.

**Comment by bentrm** [comment]
The issue has the status flag of "needs info". Is there any more info I can add to help apart from a pull request?

I just checked the database and can find the following outcome in the text block of that page (a stream field with a rich text component in this case).

````
<a id="1" linktype="page">www.google.de</a>
````

I removed some escaping and changed the URL that has been tried to link to. I think it was meant to be an external link and the user misinterpreted the meaning of "root" in the interface.
